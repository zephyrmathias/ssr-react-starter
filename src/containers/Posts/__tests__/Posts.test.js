import React from 'react';
import { shallow } from 'enzyme';
import { Posts } from '../index';

describe('Containers: <Posts />', () => {
  test('should render correctly <Posts />', () => {
    const props = {
      posts: {
        userId: 1,
        id: 2,
        title: 'title',
        body: 'body',
      },
      fetchPosts: jest.fn(),
    };
    const wrapper = shallow(<Posts {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should not re render component if nextProps is equal to oldProps', () => {
    const props = {
      posts: {
        userId: 1,
        id: 2,
        title: 'title',
        body: 'body',
      },
      fetchPosts: jest.fn(),
    };
    const wrapper = shallow(<Posts {...props} />);
    const shouldUpdate = wrapper.instance().shouldComponentUpdate(props);
    expect(shouldUpdate).toBe(false);
  });

  test('should return an array of dispatched actions once call static fetchData', () => {
    const props = {
      posts: {
        userId: 1,
        id: 2,
        title: 'title',
        body: 'body',
      },
      fetchPosts: jest.fn(),
    };
    const wrapper = shallow(<Posts {...props} />);
    const s = wrapper;
    console.log(s);
  });
});
