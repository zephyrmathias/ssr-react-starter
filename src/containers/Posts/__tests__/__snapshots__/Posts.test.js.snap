// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers: <Posts /> should render correctly <Posts /> 1`] = `
<div>
  <h1>
    POST
  </h1>
  <h2>
    UserId: 
    1
  </h2>
  <h4>
    Id: 
    2
  </h4>
  <h4>
    Title: 
    title
  </h4>
  <p>
    body
  </p>
</div>
`;
